### PR TITLE
TLS test failures

### PR DESCRIPTION
**Pull request by mrocklin**
Do not merge

### CODE REVIEW

**Comment by mrocklin** [comment]
@jcrist @mariusvniekerk do either of you have time to check out these test failures?  They're breaking tests currently.

**Comment by mrocklin** [comment]
Also cc @pitrou

**Comment by pitrou** [comment]
You should be able to pinpoint when the failures started to occur? Is it on a specific changeset?

**Comment by mrocklin** [comment]
I haven't tried doing this yet.  They didn't show up during any PR and now they fail consistently, so I don't think that they snuck in as some intermittent failures do. 

My guess is that some dependency changed upstream.

**Comment by pitrou** [comment]
Do you `conda list` as part of CI jobs? That's very useful for cases like this, so that you can compare versions.

**Comment by pitrou** [comment]
By the way, I don't think those failures are intermittent. They should even be reproducible locally, with the right software setup.

**Comment by mariusvniekerk** [comment]
@pitrou Pretty certain these are related to openssl 1.1 vs 1.0

**Comment by pitrou** [comment]
@mariusvniekerk do you have a proof of that? e.g. `conda list` output?

**Comment by mariusvniekerk** [comment]
it does "warn" about it failing, so it is aware of that but doesn't raise?

**Comment by pitrou** [comment]
Right, but it should raise in that case. The warning is in the case the handshake error is because of transient network issues. But if the certificate is reject, that's a permanent error and it should raise an exception. So you must inspect the exception to decide what to do with it.

**Comment by mariusvniekerk** [comment]
We should change the test to catch and reraise the exception if present to see what it is,  we could still be raising **something** but pytest just says it's not `EnvironmentError`.  Want to just rule that one out first.
I'm not sure if i have push rights to this pr

**Comment by mrocklin** [comment]
Marius, you now have push rights to this PR

**Comment by mrocklin** [comment]
Tests seem to pass.  The Python 2 failure was due to the `print(..., flush=True)` syntax in debug code.

**Comment by mrocklin** [comment]
To the extent that I'm able to judge this seems fine to me.  There were a couple of test failures but I think that they are intermittent and unrelated.

@mariusvniekerk is there any additional work that you think should be done here?

File: distributed/comm/tcp.py [hunk 1]
```diff
@@ -329,6 +333,13 @@ def connect(self, address, deserialize=True, **connection_args):
             stream = yield client.connect(ip, port,
                                           max_buffer_size=MAX_BUFFER_SIZE,
                                           **kwargs)
+            # Under certain circumstances tornado will have a closed connnection with an error and not raise
```
[the remark this answers is not in the document]
Comment (mariusvniekerk) [reply distributed/comm/tcp.py]: @mrocklin I think this gets us working for now.
@pitrou I don't think tornado is tested a python built against openssl 1.1+ atm?

[the remark this answers is not in the document]
Comment (pitrou) [reply distributed/comm/tcp.py]: I don't know, but it's still worth reporting. I generally find it annoying when people don't report issues upstream.

[the remark this answers is not in the document]
Comment (mariusvniekerk) [reply distributed/comm/tcp.py]: Done

**Comment by mrocklin** [comment]
I removed my initial dummy commit and added a couple small style changes.  I hope that now that the first commit belongs to @mariusvniekerk that the "Squash and merge" feature will appropriately assign ownership to him

**Comment by mrocklin** [comment]
This does mean a force push though, so you may want to `git reset --hard` before pulling/pushing if there is any more to do

**Pull request merged**